### PR TITLE
Changes for ISV to get this information using K8s calls.

### PR DESCRIPTION
**Pull request by dlabaj**
Resolves: #190 

Contains the changes needed to read ISV from k8s calls instead of using the yaml files in the repo.

- [x] Backend update
- [x] yaml files removed
- [x] Frontend updated

Dependent on https://github.com/red-hat-data-services/odh-deployer/pull/260

### CODE REVIEW

File: backend/src/utils/resourceUtils.ts [hunk 1]
```diff
@@ -133,93 +148,157 @@ const fetchInstalledKfdefs = async (fastify: KubeFastifyInstance): Promise<KfDef
...
-  // fetch enabled apps configmap
-  const coreV1Api = fastify.kube.coreV1Api;
-  const namespace = fastify.kube.namespace;
-  const enabledAppsCM: V1ConfigMap = await coreV1Api
-    .readNamespacedConfigMap(enabledAppsConfigMapName, namespace)
-    .then((result) => result.body)
-    .catch(() => null);
+  const customObjectsApi = fastify.kube.customObjectsApi;
+  try {
+    const res = await customObjectsApi.listNamespacedCustomObject(
```
Comment (jeff-phillips-18) [inline backend/src/utils/resourceUtils.ts:156]: There will be an issue if there are more than 250 applications. We should fetch in a loop, similar to fetching subscriptions: https://github.com/opendatahub-io/odh-dashboard/blob/main/backend/src/utils/resourceUtils.ts#L75

File: backend/src/utils/resourceUtils.ts [hunk 2]
```diff
@@ -133,93 +148,157 @@ const fetchInstalledKfdefs = async (fastify: KubeFastifyInstance): Promise<KfDef
...
-        if (doc.spec.featureFlag) {
-          if (featureFlags[doc.spec.featureFlag]) {
-            docs.push(doc);
-          }
-          return;
-        }
-        if (!doc.spec.appName || appDefs.find((def) => def.metadata.name === doc.spec.appName)) {
+  try {
+    const customObjectsApi = fastify.kube.customObjectsApi;
+    const res = await customObjectsApi.listNamespacedCustomObject(
```
Comment (jeff-phillips-18) [inline backend/src/utils/resourceUtils.ts:206]: Same problem here, should fetch in a loop.

File: backend/src/routes/api/components/list.ts [hunk 3]
```diff
@@ -1,34 +1,18 @@
...
 ): Promise<OdhApplication[]> => {
-  const applicationDefs = getApplicationDefs();
-  const installedComponents = [];
+  const applications = await getApplications();
   const query = request.query as { [key: string]: string };
 
-  if (!query.installed) {
-    return await Promise.all(applicationDefs);
+  if (query.installed) {
+    return Promise.resolve(applications.filter((app) => app.spec.isEnabled));
```
Comment (jeff-phillips-18) [inline backend/src/routes/api/components/list.ts:13]: We need to determine if the application was disabled by the user at some previous point and not return those that were.

**Review by andrewballantyne — COMMENTED** [review]
Few nits -- nothing critical.

Question about a couple things though:
1. How does the cluster get bootstrapped with the CR items (and the CRD `odhquickstarts`); they are no longer part of the standard deployment (as we were file reading from local files before) -- so there is a deployment change that needs to be handled "here"
2. I know it's not quickstarts, but any thoughts on getting rid of the last file read ([features.ts](https://github.com/opendatahub-io/odh-dashboard/blob/main/backend/src/utils/features.ts#L7))?

You also have a lot of conflicts... so this will need a rebase. Not sure if we gate PRs on failed lints, but you have a few of those too.

File: backend/src/utils/resourceUtils.ts (hunk 1 above)
Comment (dlabaj) [reply backend/src/utils/resourceUtils.ts:156]: done.

File: backend/src/utils/resourceUtils.ts (hunk 2 above)
Comment (dlabaj) [reply backend/src/utils/resourceUtils.ts:206]: done.

File: backend/src/utils/resourceUtils.ts [hunk 4]
```diff
@@ -133,93 +148,157 @@ const fetchInstalledKfdefs = async (fastify: KubeFastifyInstance): Promise<KfDef
...
+  return Promise.resolve(updatedDocApps);
+};
+
+const fetchQuickStarts = async (fastify: KubeFastifyInstance): Promise<QuickStart[]> => {
+  const installedQuickStarts: QuickStart[] = [];
+  const featureFlags = getComponentFeatureFlags();
+
+  const customObjectsApi = fastify.kube.customObjectsApi;
+  try {
+    const res = await customObjectsApi.listNamespacedCustomObject(
```
[the remark this answers is not in the document]
Comment (dlabaj) [reply backend/src/utils/resourceUtils.ts:285]: done.

**Comment by andrewballantyne** [comment]
Putting a hold so this does not merge for this release and waits until next week. This is too risky for 1.16 of RHODS which follows up on our 2.1 ODH release this Friday.

**Comment by DaoDaoNoCode** [comment]
I suspect we could safely remove the `data` folder here in this PR since we don't know how the `OdhApplication`, `OdhQuickStart` and `OdhDocument` CRs are created. Now all the YAML definitions are moved there on the cluster. We need to make sure the operator is creating everything correctly when deploying the opendatahub before we merge the PR.

**Comment by LaVLaS** [comment]
> I suspect we could safely remove the `data` folder here in this PR since we don't know how the `OdhApplication`, `OdhQuickStart` and `OdhDocument` CRs are created. Now all the YAML definitions are moved there on the cluster. We need to make sure the operator is creating everything correctly when deploying the opendatahub before we merge the PR.

For the new CRs that were previously stored under `data/` are you re-creating those CRs under in the `manifests/` directory?

**Comment by DaoDaoNoCode** [comment]
@LaVLaS No, so do you mean we need to actually move those resources to the `manifests` repo as well as the CRD?

File: backend/src/routes/api/components/list.ts (hunk 3 above)
Comment (dlabaj) [reply backend/src/routes/api/components/list.ts]: done.

**Review by andrewballantyne — APPROVED** [review]

**Pull request merged**